design/proof-erasure-coding.md

```diff
@@ -0,0 +1,244 @@
+            \__/                        \     /
+                    ^                    \   /       |
+                    |                     ---        |
+     ^  ^           |  ^        |                    |
+     |  |        ^  |  |  ^     |  |  |           |  |
+     |  |  ^     |  |  |  |     |  |  |        |  |  |
+     |  |  |  ^  |  |  |  |     |  |  |  |  |  |  |  |
+     |  |  |  |  |  |  |  |     |  |  |  |  |  |  |  |
+     |  |  |  |  |  |  |  |     v  v  v  v  v  v  v  v
+
+    -------------------------  -------------------------
+    |//|//|//|//|//|//|//|//|  |//|//|//|//|//|//|//|//|
+    -------------------------  -------------------------
+
+         original data                  parity
+
+
+This only works for small amounts of data. When the polynomial is for instance
+defined over byte sized elements from a Galios field of 2^8, you can only encode
+2^8 = 256 bytes (data and parity combined).
+
+Interleaving
+------------
+
+To encode larger pieces of data with erasure coding, interleaving is used. This
+works by taking larger blocks of data, and encoding smaller elements from these
+blocks.
+
+    data blocks
+
+    -------------    -------------    -------------    -------------
+    |x| | | | | |    |x| | | | | |    |x| | | | | |    |x| | | | | |
+    -------------    -------------    -------------    -------------
+     |                /                /                |
+      \___________   |   _____________/                 |
+                  \  |  /  ____________________________/
+                   | | |  /
+                   v v v v
+
+                  ---------         ---------
+            data  |x|x|x|x|   -->   |p|p|p|p|  parity
+                  ---------         ---------
+
+                                     | | | |
+       _____________________________/ /  |  \_________
+      /                 _____________/   |             \
+     |                 /                /               |
+     v                v                v                v
+    -------------    -------------    -------------    -------------
+    |p| | | | | |    |p| | | | | |    |p| | | | | |    |p| | | | | |
+    -------------    -------------    -------------    -------------
+
+    parity blocks
+
+This is repeated for each element inside the blocks.
+
+Adversarial erasure
+-------------------
+
+The disadvantage of interleaving is that it weakens the protection against
+adversarial erasure that Reed-Solomon provides.
+
+An adversary can now strategically remove only the first element from more than
+half of the blocks, and the dataset will be damaged beyond repair. For example,
+with a dataset of 1TB erasure coded into 256 data and parity blocks, an
+adversary could strategically remove 129 bytes, and the data can no longer be
+fully recovered.
+
+Implications for storage proofs
+-------------------------------
+
+This means that when we check for missing data, we should perform our checks on
```

Comment:
What do you mean by blocks in this context?

Reply:
Blocks of interleaved data, of which there are N. I'll clarify the text.